linthesia: init at 0.8.0

ckiee opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Playing piano better!

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

ckiee committed:
linthesia: init at 0.8.0

vcunat commented on pkgs/games/linthesia/default.nix:
```diff
@@ -0,0 +1,38 @@
+
+  meta = with lib; {
+    description = "A game of playing music using a MIDI keyboard following a MIDI file";
+    homepage = "http://linthesia.sourceforge.net";
```
Nit: is this homepage really a good choice?  By looking at it I'd think the SW has been abandoned for 10+ years.  I'd suggest `inherit (src.meta) homepage;` instead.

vcunat commented on pkgs/games/linthesia/default.nix:
```diff
@@ -0,0 +1,38 @@
+    repo = "linthesia";
+    rev = "11669e355c72fefefb2b3adee31e821dc4c4c9ce";
+    sha256 = "sha256-FmzQjd9rR9NS8za/kgdCACaEqk2J5KpYrjInC0q9Ep4=";
+  };
```
There's been a release in the meantime.  Perhaps use that instead?  https://github.com/linthesia/linthesia/releases/tag/0.8.0

vcunat reviewed (commented):
I'm not sure if it's just me, but rendered text looks like unreadable noise instead.

ckiee commented:
> I'm not sure if it's just me, but rendered text looks like unreadable noise instead.

![A screenshot showing two windows, on the left a terminal emulator running `./result/bin/linthesia` and on the right the linthesia program running, showing a list of folders: favorites, game_themes, hannon, keyboard_classics, learning, popular.](https://user-images.githubusercontent.com/25263210/152720474-4395d2fd-98ea-4210-b950-b2bac26b7299.png)

(as for the review comments) Here I was thinking I could get away with a perfect PR this time around, but alas; fixed it now :P


[Now for the obligatory ping in case GitHub decides sending notifications is bad: @vcunat]

vcunat commented:
It shows OK on _your_ commit, but when I merge that to current master I get
![linthesia-noise](https://user-images.githubusercontent.com/1785925/152749660-6fe57774-7513-4863-a542-29f5c3270e37.png)

vcunat commented:
I wonder if this log is related to the noise:
```
(process:2388069): GLib-GObject-CRITICAL **: 09:11:42.214: g_object_set_qdata_full: assertion 'quark > 0' failed
```

Anyway, I suggest you try the commit atop current nixpkgs master as well.

ckiee committed:
SDL2_ttf_2_0_15: init

This is an older version of the SDL2_ttf derivation needed for
`linthesia` to run properly. I don't know why a minor version change
broke things (2.0.15 -> 2.0.18) but it did and this fixes that without
disrupting other packages.

ckiee commented:
> Anyway, I suggest you try the commit atop current nixpkgs master as well.

Rebasing reproduced your screenshot instantly and the culprit appears to be #154827 as reverting that commit fixes it, haven't pushed the revert since I can't reproduce this issue on any other dependants of `SDL2_ttf`.

vcunat reviewed (approved):
OK, assuming it work well for you overall.

ckiee commented:
![A screenshot practically identical to the previous one in this PR](https://user-images.githubusercontent.com/25263210/152850169-401743f3-5334-4b0e-8438-c2017847c9fc.png)

Merge, pretty please?

McSinyx commented:
Have you contacted upstream about compatibility with SDL_ttf 2.0.18?  Unless it is NixOS-specific problem with its integration, I don't think rushing inclusion of linthesia with SDL_ttf 2.0.15 is the right thing to do here and linthesia should be fixed instead.

vcunat commented:
I believe it's OK to merge this makeshift solution.

SuperSandro2000 commented on pkgs/development/libraries/SDL2_ttf/2.0.15.nix:
```diff
@@ -0,0 +1,25 @@
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ SDL2 freetype libGL ]
+    ++ lib.optional stdenv.isDarwin darwin.libobjc;
```
darwin shouldn't be used as an input like pkgs.

vcunat replied:
I agree, but that's a problem in the `SDL2_ttf` expressions, in both versions.  Not really in this PR.
